src/chapter16/Checkbox.java

```diff
@@ -0,0 +1,138 @@
+package chapter16;
+
+/*
+  Based on Try This 16-1
+  
+  File comparer with a check box
+
+*/
+
+import java.awt.*;
+import java.awt.event.*;
+import javax.swing.*;
+import java.io.*;
+
+public class Checkbox implements ActionListener {
+
+  JTextField jtfFirst;  // holds the first file name
+  JTextField jtfSecond; // holds the second file name
+
+  JButton jbtnComp; // button to compare the files
+
+  JLabel jlabFirst, jlabSecond; // displays prompts
+  JLabel jlabResult; // displays results and error messages
+
+  JCheckBox checkbox;
+  JLabel checkboxResult;
+
+  Checkbox() {
+
+   // Create a new JFrame container.
+   JFrame jfrm = new JFrame("Compare Files");
+
+   // Specify FlowLayout for the layout manager.
+   jfrm.setLayout(new FlowLayout());
+
+   // Give the frame an initial size.
+   jfrm.setSize(200, 250);
+
+   // Terminate the program when the user closes the application.
+   jfrm.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+   // Create the text fields for the file names..
+   jtfFirst = new JTextField(14);
+   jtfSecond = new JTextField(14);
+
+   // Set the action commands for the text fields.
+   jtfFirst.setActionCommand("fileA");
+   jtfSecond.setActionCommand("fileB");
+
+   // Create the Compare button.
+   JButton jbtnComp = new JButton("Compare");
+
+   // Add action listener for the Compare button.
+   jbtnComp.addActionListener(this);
+   
+   // Creating a checkbox
+   checkbox = new JCheckBox( "Show Mismatch?" );
+   checkbox.setActionCommand( "show_mismatch" );
+
+   // Create the labels.
+   jlabFirst = new JLabel("First file: ");
+   jlabSecond = new JLabel("Second file: ");
+   jlabResult = new JLabel("");
+   checkboxResult = new JLabel("");
+
+   // Add the components to the content pane.
+   jfrm.add(jlabFirst);
+   jfrm.add(jtfFirst);
+   jfrm.add(jlabSecond);
+   jfrm.add(jtfSecond);
+   jfrm.add(checkbox);
+   jfrm.add(jbtnComp);
+   jfrm.add(jlabResult);
+   jfrm.add(checkboxResult);
```

Comment:
I wish `JFrame` had a variadic version of `add` to reduce this type of boilerplate. In modern Java it'd be a two line method (or one line if you hate operator chaining):
```java
public JFrame add(Component... components) {
  Arrays.stream(components).forEachOrdered(this::add);
  return this;
}
```